use time.perf_counter and time.process_time

rasbt opened the pull request:
Adds an optional argument `include_sleep` to count time elapsed including sleep time (`include_sleep=True`) and excluding sleep time (`include_sleep=False`) (addressing #7)

Note: I added a Python 3.6 badge to the Readme; haven't tested it on Python 2.7

ramonsaraiva commented on timy/__init__.py:
```diff
@@ -25,36 +25,52 @@ def output(ident, text):
         self.ident = ident
         self.start = 0
+        self.end = 0
+        self.include_sleep = include_sleep
```
What about holding the function in a variable and then calling it instead of using the if later?
Like: `self.time = time.perf_counter if include_sleep else time.process_time`

ramonsaraiva commented on timy/__init__.py:
```diff
@@ -25,36 +25,52 @@ def output(ident, text):
 
     def __enter__(self):
-        self.start = time.time()
+        if self.include_sleep:
```
Then here we can use `self.time()`

ramonsaraiva commented on timy/__init__.py:
```diff
@@ -25,36 +25,52 @@ def output(ident, text):
     @property
     def elapsed(self):
-        return time.time() - self.start
+        if self.include_sleep:
```
And here we can also use `self.time()`

ramonsaraiva commented on timy/__init__.py:
```diff
@@ -25,36 +25,52 @@ def output(ident, text):
 
 
-def timer(ident=timy_config.DEFAULT_IDENT, loops=1):
+def timer(ident=timy_config.DEFAULT_IDENT, loops=1, include_sleep=True):
```
We can apply the same idea here, having a variable holding the correct function depending on the parameter. And then replace both ifs with it.

ramonsaraiva commented on README.md:
```diff
@@ -1,5 +1,7 @@
 # timy
 
+![Python 3.6](https://img.shields.io/badge/python-3.6-blue.svg)
```
👍

rasbt commented:
Good points. I changed it so that it only checks what `time` function to use once in the beginning -- to avoid unnecessary if-else clauses. Also, I noticed that `'seconds'` got printed twice: once by the `wrapper` function in `timer` and once by the `output` function, resulting in `'seconds seconds'`.  Thus, I removed the one `seconds` from the `wrapper` func.

ramonsaraiva commented:
Thanks, merging it.
